lib/stack/GradleBuildInterpreter.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright © 2019 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    AutofixRegistration,
+    CodeInspectionRegistration,
+    goals,
+    isMaterialChange,
+} from "@atomist/sdm";
+import { Version } from "@atomist/sdm-core";
+import {
+    AutofixRegisteringInterpreter,
+    CodeInspectionRegisteringInterpreter,
+    Interpretation,
+    Interpreter,
+} from "@atomist/sdm-pack-analysis";
+import { Build } from "@atomist/sdm-pack-build";
+import { DockerBuild } from "@atomist/sdm-pack-docker";
+import {
+    gradleBuilder,
+} from "@atomist/sdm-pack-spring";
+import {
+    GradleBuild,
+    GradleDefaultOptions,
+    GradleProjectVersioner,
+    GradleVersion,
+} from "@atomist/sdm-pack-spring/lib/gradle/build/helpers";
+import { HasVersionDefinedInGradleProperties } from "@atomist/sdm-pack-spring/lib/gradle/review/hasGradleVersion";
+import { BuildSystemStack } from "./buildSystemScanner";
+
+/**
+ * Interpreter that adds a gradle build goal when Gradle is found in the project's interpretation.
+ * @see buildSystemScanner
+ */
+export class GradleBuildInterpreter implements Interpreter, AutofixRegisteringInterpreter, CodeInspectionRegisteringInterpreter {
+
+    // This includes test goal
+    private readonly gradleBuildGoal: Build = new Build()
+        .with({
+            ...GradleDefaultOptions,
+            builder: gradleBuilder(),
+        });
+
+    private readonly gradleVersionGoal: Version = new Version()
+        .with({
+            ...GradleDefaultOptions,
+            versioner: GradleProjectVersioner,
+        });
+
+    private readonly dockerBuildGoal: DockerBuild = new DockerBuild()
+        .with({
```

Comment:
progress reporter and dockerfile finder missing

Reply:
Fixed